StatsdClient/Statsd.cs

```diff
@@ -116,8 +140,10 @@ private void InitialiseInternal(Func<IOutputChannel> createOutputChannel, string
         /// <param name="name">The metric name.</param>
         /// <param name="milliseconds">The duration, in milliseconds, for this metric.</param>
         public async Task LogTimingAsync(string name, long milliseconds)
-        {
-            await SendMetricAsync(MetricType.TIMING, name, _prefix, milliseconds);
+        {
+            if (milliseconds < 0)

```

Comment:
I noticed that in the method public async Task LogCountAsync(string name, long count = 1) the code is validating that the name is not null. Should we not be doing this in all methods?